control_plane/src/compute.rs

```diff
@@ -205,38 +203,48 @@ impl PostgresNode {
             env: env.clone(),
             pageserver: Arc::clone(pageserver),
             is_test: false,
+            timelineid
         })
     }
 
-    fn init_vanilla(&self) -> Result<()> {
+    // Connect to a page server, get base backup, and untar it to initialize a
+    // new data directory
+    pub fn init_from_page_server(&self) -> Result<()> {
+
+        let pgdata = self.pgdata();
+
         println!(
-            "Creating new postgres: path={} port={}",
-            self.pgdata().to_str().unwrap(),
+            "Extracting base backup to create postgres instance: path={} port={}",
+            pgdata.to_str().unwrap(),
             self.address.port()
         );
 
         // initialize data directory
-
         if self.is_test {
-            fs::remove_dir_all(self.pgdata().to_str().unwrap()).ok();
+            fs::remove_dir_all(&pgdata).ok();
         }
 
-        fs::create_dir_all(self.pgdata().to_str().unwrap())?;
+        let sql = format!("basebackup {}", self.timelineid);
+        let mut client = self.pageserver.page_server_psql_client()?;
+        println!("connected to page server");
 
-        let initdb_path = self.env.pg_bin_dir().join("initdb");
-        let initdb = Command::new(initdb_path)
-            .args(&["-D", self.pgdata().to_str().unwrap()])
-            .arg("-N")
-            .arg("-A trust")
-            .arg("--no-instructions")
-            .env_clear()
-            .env("LD_LIBRARY_PATH", self.env.pg_lib_dir().to_str().unwrap())
-            .stdout(Stdio::null())
-            .status()?;
+        fs::create_dir_all(&pgdata)?;
+        fs::set_permissions(pgdata.as_path(), fs::Permissions::from_mode(0o700)).unwrap();
 
-        if !initdb.success() {
-            return Err("initdb failed".into());
-        }
+        // Also create pg_wal directory, it's not included in the tarball
+        // FIXME: actually, it is currently.
+        //fs::create_dir_all(pgdata.join("pg_wal"))?;
```

Comment:
I'm not really following these comments. Are they saying this line of code can just go away?